server/mediapublic/mediapublic/views.py

```diff
@@ -0,0 +1,686 @@
+
+from sqlalchemy.exc import DBAPIError
+
+from .models import (
+    DBSession,
+    Users,
+    UserTypes,
+    RecordingCategories,
+    Comments,
+    Organizations,
+    People,
+    Recordings,
+    Howtos,
+    Blogs,
+    Playlists,
+    PlaylistAssignments,
+    )
+
+import datetime
+import json
+
+def check_payload(body, cls, keys=[]):
+
+    dt_objs = [
+        'creation_datetime',
+        'edit_datetime',
+        'recorded_datetime',
+    ]
+
+    payload = {}
+    resp = None
+    #code = 200
+    try:
+        payload = json.loads(body)
+        if not isinstance(payload, dict):
+            payload = {}
+            resp = {'error': 'JSON not dictionary.'}
+            #code = 400
+        else:
+            reqkeys = cls.reqkeys()
+            if not ( (len(reqkeys) + len(keys)) == len(payload) and \
+                    all(k in payload for k in reqkeys) and all(k in payload for k in keys) ):
+                payload = {}
+                resp = {'error': 'Missing Key.  Required Keys: %s, %s' % (cls.reqkeys(), keys) }
+                #code = 400
+            else:
+                for k in payload:
+                    if k in dt_objs:
+                        try:
+                            payload[k] = datetime.datetime.strptime(payload[k], "%Y-%m-%d %H:%M:%S.%f")
+                        except:
+                            payload = {}
+                            resp = {'error': 'Invalid DateTime format for key: %s' % k}
+                            break
+    except:
+        resp = {'error': 'Invalid JSON'}
+    #    #code = 400
+
+    return payload, resp #, code
+
+def do_post(request, keys, cls,):
+    resp = {'error': 'Invalid Input'}
+    status = 400
+    payload, err = check_payload(request.body, cls)
+    if err is None:
+        for k in keys:
+            payload[k] = keys[k]
+        payload['creation_datetime'] = datetime.datetime.now()
+        org = cls.add(**payload)
+        resp = org.to_dict()
+        status = 201
+    else:
+        resp = err
+    return resp, status
+
+def do_get_single(request, id, cls):
+    resp = {'error': 'Not Found'}
+    status = 404
+    thing = cls.get_by_id(id)
+    if not thing is None:
+        resp = thing.to_dict()
+        status = 200
+    return resp, status
+
+def do_put(request, id, cls):
+
+    resp = {'error': 'Not Found'}
+    status = 404
+    payload, err = check_payload(request.body, cls)
+    if err is None:
+        thing = cls.update_by_id(id, **payload)
+        if not thing is None:
+            resp = thing.to_dict()
+            status = 201
+    else:
+        resp = err
+    return resp, status
+
+def do_delete(request, id, cls):
+    resp = {'error': 'Not Found'}
+    status = 404
+    thing = cls.delete_by_id(id)
+    if not thing is None:
+        resp = {}
+        status = 204
+    return resp, status
+
+METHODS = {
+    'GET': do_get_single,
+    #'POST': do_post,
+    'PUT': do_put,
+    'DELETE': do_delete,
+}
```

Comment:
One potential suggestion:
1. Rename this SINGLE_METHODS, or something (and remove POST comment).
2. Make an analogous ROOT_METHODS (or something) with just POST and GET (GET pointing to a not-yet-existent do_get_all).
3. Change up the /resources routes to look more like the /resources/:id routes, but using the ROOT_METHODS.